cmd/goal/clerk.go

```diff
@@ -144,9 +147,10 @@ func init() {
 	dryrunRemoteCmd.Flags().BoolVarP(&rawOutput, "raw", "r", false, "Output raw response from algod")
 	dryrunRemoteCmd.MarkFlagRequired("dryrun-state")
 
-	simulateCmd.Flags().StringVarP(&txFilename, "txfile", "t", "", "Transaction or transaction-group to test")
+	simulateCmd.Flags().StringVarP(&txFilename, "txfile", "t", "", "Transaction or transaction-group to test. Mutually exclusive with --request")
+	simulateCmd.Flags().StringVar(&requestFilename, "request", "", "Simulate request object to run. Mutually exclusive with --txfile")
+	simulateCmd.Flags().StringVar(&requestOutFilename, "request-out", "", "Filename for writing simulate request object. If provided, the command will only write the request object and exit. No simulation will happen")
 	simulateCmd.Flags().StringVarP(&outFilename, "outfile", "o", "", "Filename for writing simulation result")
```

Comment:
`result-out`?

Reply:
`response-out`?

Reply:
`request-out` is intentional. This flag indicates you want goal to build and save the request, but not send it.

Reply:
sorry this must have been updated while I was editing, I meant, "should outfile be result-out"

Reply:
Changed in 5b012f313